docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,134 @@ The names of the functions are also available as the keys of the
+                    follow: ``tile``, ``foreground color``, ``background color``, ``bright``,
+                    ``graphics tile``, ``overlay tile``, ``signpost tile``, ``item tile``.
+                    First 4 are function same as ascii workshop definition. The latter 4 are graphics
+                    layers. ``signpost tile`` is an optional row that sticks up over the workshop.
 
-    :name:
-        custom workshop id e.g. ``SOAPMAKER``
+  :frame_skip:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                    are synchronized with machine animation.
 
-        .. note:: this is the only mandatory field.
+* ``setAnimationInfoAuto(workshop_type, make_graphics_too, [frame_length], [gear_tiles])``
 
-    :fix_impassible:
-        if true make impassable tiles impassable to liquids too
-    :consume:
-        how much machine power is needed to work.
-        Disables reactions if not supplied enough and ``needs_power==1``
-    :produce:
-        how much machine power is produced.
-    :needs_power:
-        if produced in network < consumed stop working, default true
-    :gears:
-        a table or ``{x=?,y=?}`` of connection points for machines.
-    :action:
-        a table of number (how much ticks to skip) and a function which
-        gets called on shop update
-    :animate:
-        a table of frames which can be a table of:
+  Animate workshop as with function above but generate frames automatically. This works by finding
+  tiles which have gears and animating them with alternating gear tiles.
 
-        a. tables of 4 numbers ``{tile,fore,back,bright}`` OR
-        b. empty table (tile not modified) OR
-        c. ``{x=<number> y=<number> + 4 numbers like in first case}``,
-           this generates full frame useful for animations that change little (1-2 tiles)
+  :workshop_type:     custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :make_graphics_too:  replace same tiles in graphics mode with tiles from vanilla df mechanism
+  :frame_length:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                      are synchronized with machine animation.
+  :gear_tiles:        Optional array of 2 or 4 indexes. First two define ascii tiles and next two graphics tiles.
+                      This overrides default gear tiles.
 
-    :canBeRoomSubset:
-        a flag if this building can be counted in room. 1 means it can, 0 means it can't and -1 default building behaviour
-    :auto_gears:
-        a flag that automatically fills up gears and animations. It looks over the building definition for gear icons and maps them.
+* ``setOnUpdate(workshop_type,interval,callback)``
 
-    Animate table also might contain:
+  Setup callback to be called every ``interval`` of ticks for each building of this type. Note: low interval
+  numbers and/or many workshops that use this might reduce DF performance.
```

Comment:
```suggestion
  Register callback to be called every ``interval`` ticks for each building of this
  type. This can be very expensive if the interval is low and/or there are many
  workshops of this type. Keep these callbacks light!
```

Reply:
you may want to explain the difference between frame ticks and world simulation ticks (and note which one this refers to).